File: Binary Search I/python/Day 4/sqrt_of_num.py
class Solution:
    def mySqrt(self, x: int) -> int:
        '''
        Simply search for squreroot as we add in perfect squre question in Study plan, Binary Search I (Day 3)
          a) if Root found return root
          b) else return smaller than root which is end pointer in this case
        '''
        s=1
        e=x
        while(s<=e):
            m=s+(e-s)//2
            mm=m*m
            if(mm==x):
                return m
            
            if(mm>x):
                e=m-1
            else:
                s=m+1
            
        
        return e
        

File: Binary Search I/java/Day 4/SqrtOfNum.java
class Solution {
    public int mySqrt(int x) {
        /**
          Simply search for squreroot as we add in perfect squre question in Study plan, Binary Search I (Day 3)
          a) if Root found return root
          b) else return smaller than root which is end pointer in this case
         */
        int s=1;
        int e=x;
        while(s<=e){
            int mid=s+(e-s)/2;
            double mm=Math.pow(mid,2);
            if(mm==x){
                return mid;
            }
            if(mm>x){
                e=mid-1;
            }else{
                s=mid+1;
            }
        }
        return e;
    }
}